pkg/engine/engine_test.go

```diff
@@ -96,3 +110,156 @@ func TestUnmarshalTaskBuild(t *testing.T) {
 		t.Errorf("Unmarshal Build task returned incorrect data")
 	}
 }
+
+func TestNewEngine(t *testing.T) {
```

Comment:
We are testing task timeout, but it can only be invoked upon calling constructor of Engine, because that is when it is popping tasks from the queue.. this was firstly intended to be integration test, but I would remove it totally to avoid confusion, since integration test is executed from the script, as you told me so.

Reply:
I would remove the tests, as well. The integration test covers the desired usage.